plugin.js

```diff
@@ -23,17 +24,25 @@ class WasmPackPlugin {
     this.crateDirectory = options.crateDirectory;
     this.forceWatch = options.forceWatch;
     this.forceMode = options.forceMode;
-    this.extraArgs = (options.extraArgs || '').trim().split(' ').filter(x=> x);
+    this.extraArgs = (options.extraArgs || '').trim().split(' ').filter(x => x);
+    this.outDir = options.outDir || "pkg";
+    this.outName = options.outName || "index";
     this.watchDirectories = (options.watchDirectories || [])
       .concat(path.resolve(this.crateDirectory, 'src'));
+    this.watchFiles = [path.resolve(this.crateDirectory, 'Cargo.toml')];
 
     this.wp = new Watchpack();
     this.isDebug = true;
+    this.error = null;
   }
 
   apply(compiler) {
     this.isDebug = this.forceMode ? this.forceMode === "development" : compiler.options.mode === "development";
 
+    // This fixes an error in Webpack where it cannot find
+    // the `pkg/index.js` file if Rust compilation errors.
+    this._makeEmpty();
```

Comment:
If you already important an non-existing module with webpack, it will tell you in the console and at runtime, It has quite a good UX IMO. We could create a file with a ThrowStatement with the compilation error in it.

Reply:
I don't think the UX is good. This is what you get with the old version:

```
error[E0599]: no function or associated item named `from_str2` found for type `wasm_bindgen::JsValue` in the current scope
  --> src\lib.rs:24:30
   |
24 |     console::log_1(&JsValue::from_str2("Hello world!"));
   |                              ^^^^^^^^^
   |                              |
   |                              function or associated item not found in `wasm_bindgen::JsValue`
   |                              help: there is a method with a similar name: `from_str`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
error: Could not compile `rust-webpack-template`.

To learn more, run the command again with --verbose.
Error: Compiling your crate to WebAssembly failed
Caused by: failed to execute `cargo build`: exited with exit code: 101
wasm-pack error: undefined
Hash: 3b93c8acec8101b6a7f9
Version: webpack 4.35.2
Time: 1818ms
Built at: 07/02/2019 2:46:07 PM
 1 asset
Entrypoint index = index.js
[0] ./js/index.js 49 bytes {0} [built]

ERROR in ./js/index.js
Module not found: Error: Can't resolve '../pkg/index.js' in 'C:\Users\Pauan\Shared Folders\NixOS\tmp\my-app\js'
 @ ./js/index.js 1:0-25
error Command failed with exit code 2.
```

This part here is useless, and it actually pushes away the useful information:

```
ERROR in ./js/index.js
Module not found: Error: Can't resolve '../pkg/index.js' in 'C:\Users\Pauan\Shared Folders\NixOS\tmp\my-app\js'
 @ ./js/index.js 1:0-25
```

So my changes gets rid of it. You still get an error message, just without the useless stuff.

Reply:
Sorry, I typed the comment too fast. I just wanted to say that we could show the error on the browser-side as well, by injecting code with a ThrowStatement in `./pkg/index.js`.

Reply:
As for putting the compilation error into a file (or whatever), I tried that but it's not possible.

When you spawn a command (like `wasm-pack build`) you have two options: you can pipe the output, or you can inherit the `stdin`/`stdout`/`stderr`.

If you pipe the output, then you can store it in a file, but then you don't get any colors from `cargo build`.

If you inherit the output, then you get nice colors with `cargo build`, but then you can't retrieve the output, so you can't put it into a file.

I tried all sorts of workarounds, nothing worked. So I gave up.